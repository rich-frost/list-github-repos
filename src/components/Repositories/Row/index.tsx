import { Tr, Td, Icon, Link } from '@chakra-ui/react';
import { GoStar, GoRepoForked } from 'react-icons/go';
import { SearchNode } from '../../../types/search';
import { BiLinkExternal } from 'react-icons/bi';

type RowProps = {
  data: SearchNode;
};

export default function Row({ data }: RowProps) {
  return (
    <Tr>
      <Td>
        {data.url ? (
          <Link href={data.url} isExternal>
            {data.name} <Icon as={BiLinkExternal} mx="2px" w={3} h={3} />
          </Link>
        ) : (
          data.name
        )}
      </Td>
      <Td>
        <Icon as={GoStar} /> {data.stargazerCount.toLocaleString()}
      </Td>
      <Td>
        <Icon as={GoRepoForked} /> {data.forkCount.toLocaleString()}
      </Td>
    </Tr>
  );
}
