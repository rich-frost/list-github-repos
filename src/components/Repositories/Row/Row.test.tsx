import renderer from 'react-test-renderer';
import Row from './index';
import { Table, Tr } from '@chakra-ui/react';

describe('Row component', () => {
  const data = {
    id: '123',
    name: 'React repo',
    stargazerCount: 100,
    forkCount: 240,
    url: 'google.com',
  };

  it('renders the component', () => {
    const component = renderer.create(
      <Table>
        <Tr>
          <Row data={data} />
        </Tr>
      </Table>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('formats large numbers appropriately', () => {
    const component = renderer.create(
      <Table>
        <Tr>
          <Row
            data={{ ...data, stargazerCount: 110023, forkCount: 629642297 }}
          />
        </Tr>
      </Table>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
