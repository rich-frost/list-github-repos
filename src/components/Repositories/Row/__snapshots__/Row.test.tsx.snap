// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component formats large numbers appropriately 1`] = `
<table
  className="chakra-table css-0"
>
  <tr
    className="css-0"
  >
    <tr
      className="css-0"
    >
      <td
        className="css-0"
      >
        <a
          className="chakra-link css-0"
          href="google.com"
          rel="noopener"
          target="_blank"
        >
          React repo
           
          <svg
            className="chakra-icon css-16o4ymj"
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m13 3 3.293 3.293-7 7 1.414 1.414 7-7L21 11V3z"
            />
            <path
              d="M19 19H5V5h7l-2-2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2v-5l-2-2v7z"
            />
          </svg>
        </a>
      </td>
      <td
        className="css-0"
      >
        <svg
          className="chakra-icon css-13otjrl"
          fill="currentColor"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 14 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74L14 6z"
            fillRule="evenodd"
          />
        </svg>
         
        110,023
      </td>
      <td
        className="css-0"
      >
        <svg
          className="chakra-icon css-13otjrl"
          fill="currentColor"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 10 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 1a1.993 1.993 0 0 0-1 3.72V6L5 8 3 6V4.72A1.993 1.993 0 0 0 2 1a1.993 1.993 0 0 0-1 3.72V6.5l3 3v1.78A1.993 1.993 0 0 0 5 15a1.993 1.993 0 0 0 1-3.72V9.5l3-3V4.72A1.993 1.993 0 0 0 8 1zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3 10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3-10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
            fillRule="evenodd"
          />
        </svg>
         
        629,642,297
      </td>
    </tr>
  </tr>
</table>
`;

exports[`Row component renders the component 1`] = `
<table
  className="chakra-table css-0"
>
  <tr
    className="css-0"
  >
    <tr
      className="css-0"
    >
      <td
        className="css-0"
      >
        <a
          className="chakra-link css-0"
          href="google.com"
          rel="noopener"
          target="_blank"
        >
          React repo
           
          <svg
            className="chakra-icon css-16o4ymj"
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m13 3 3.293 3.293-7 7 1.414 1.414 7-7L21 11V3z"
            />
            <path
              d="M19 19H5V5h7l-2-2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2v-5l-2-2v7z"
            />
          </svg>
        </a>
      </td>
      <td
        className="css-0"
      >
        <svg
          className="chakra-icon css-13otjrl"
          fill="currentColor"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 14 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74L14 6z"
            fillRule="evenodd"
          />
        </svg>
         
        100
      </td>
      <td
        className="css-0"
      >
        <svg
          className="chakra-icon css-13otjrl"
          fill="currentColor"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 10 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 1a1.993 1.993 0 0 0-1 3.72V6L5 8 3 6V4.72A1.993 1.993 0 0 0 2 1a1.993 1.993 0 0 0-1 3.72V6.5l3 3v1.78A1.993 1.993 0 0 0 5 15a1.993 1.993 0 0 0 1-3.72V9.5l3-3V4.72A1.993 1.993 0 0 0 8 1zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3 10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3-10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
            fillRule="evenodd"
          />
        </svg>
         
        240
      </td>
    </tr>
  </tr>
</table>
`;
